### PR TITLE
Implemented: logic to fetch groups and its processes (#7)

### PR DESCRIPTION
**Pull request by Ritika-Patel08**
### Related Issues

Related Issue #7 

### Short Description and Why It's Useful

- fetched the process groups and its related processes.

### Screenshots of Visual Changes before/after (If There Are Any)

image

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/napita-flow-manager#contribution-guideline)

### CODE REVIEW

File: src/components/Tabs.vue [hunk 1]
```diff
@@ -2,8 +2,12 @@
   <ion-page>
     <ion-tabs>
       <ion-router-outlet></ion-router-outlet>
-      <ion-tab-bar slot="bottom" v-if="showFooter()">
-        <ion-tab-button tab="more" href="/tabs/settings">
+      <ion-tab-bar slot="bottom">
+        <ion-tab-button tab="process" href="/tabs/process">
+          <ion-icon :icon="personOutline" />
+          <ion-label>{{ translate("Process") }}</ion-label>
```
Comment (amansinghbais) [inline src/components/Tabs.vue]: Change tab name to "Group"

File: src/services/GroupService.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { useUserStore } from '@/store/user'
+import api from "@/api"
+
+const FetchProcessGroups = async (root: any): Promise<any> => {
```
Comment (amansinghbais) [inline src/services/GroupService.ts]:
```suggestion
const fetchProcessGroups = async (root: any): Promise<any> => {
```

File: src/services/GroupService.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          runningCount: group.runningCount,
+          stoppedCount: group.stoppedCount,
+          invalidCount: group.invalidCount,
+          disabledCount: group.disabledCount,
+          inputPortCount: group.inputPortCount,
+          outputPortCount: group.outputPortCount
+        }));
+    return Promise.resolve(processGroupDetails);
+  } catch (err) {
+    console.log(err);
```
Comment (amansinghbais) [inline src/services/GroupService.ts]: Use logger instead of using console. Also check for other places

File: src/store/groups.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { defineStore } from "pinia";
+import { GroupService } from "@/services/GroupService"
+
+export const useGroupStore = defineStore('groups', {
+    state: () => {
+      return {
+        processGroups: [],
+        processByGroups: {} as any,
+        currentProcessGroup: {},
+				root:'a3229852-0187-1000-8abc-9e42ca1734f2'
```
Comment (amansinghbais) [inline src/store/groups.ts]: Instead of defining root here, hardcode it in the fetchProcessGroup action.

File: src/store/groups.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      getCurrentProcessGroup(state) {
+        return state.currentProcessGroup
+      },
+      getProcessByGroups(state) {
+        return state.processByGroups
+      }
+    },
+    actions: {
+      async fetchProcessGroups() {
+				const processGroups = await GroupService.FetchProcessGroups(this.root);
```
Comment (amansinghbais) [inline src/store/groups.ts]: Improve Indentation.

File: src/theme/variables.css [hunk 6]
```diff
@@ -279,6 +279,38 @@ http://ionicframework.com/docs/theming/ */
...
+.empty-state {
+  max-width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 10px;
+}
+
+.empty-state > img {
```
Comment (amansinghbais) [inline src/theme/variables.css]: Use empty state css from the apps-theme

File: src/views/groups.vue [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+</template>
+
+<script setup lang="ts">
+import { IonContent, IonHeader, IonMenuButton, IonPage, IonTitle, IonToolbar, onIonViewWillEnter } from "@ionic/vue";
+import { translate } from "../i18n"
+import { useGroupStore } from '../store/groups';
+import { computed, ref } from "vue";
+
+const groupStore = useGroupStore();
+let CurrentProcessGroup = ref('') as any;
```
Comment (amansinghbais) [inline src/views/groups.vue]: Check whether there is a need to define it using ref. If data is fetched computed property.

File: src/views/groups.vue [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  const selectedProcessGroupId = group.id
+  const selectedProcessGroup = processGroups.find(group => group.id === selectedProcessGroupId);
+
+  groupStore.setCurrentProcessGroup(selectedProcessGroup);
+  // Fetch and display process details for the selected group
+  await fetchProcessByGroups(selectedProcessGroupId); 
+}
+
+onIonViewWillEnter(async () => {
+  await fetchProcessGroups();
```
Comment (amansinghbais) [inline src/views/groups.vue]: Call action here directly, instead of calling a method.

File: src/components/Tabs.vue [hunk 9]
```diff
@@ -15,15 +19,8 @@
 <script setup lang="ts">
 import { translate } from "../i18n";
 import { IonIcon, IonLabel, IonPage, IonRouterOutlet, IonTabBar, IonTabButton, IonTabs } from "@ionic/vue";
-import { settingsOutline } from "ionicons/icons";
-import { useRouter } from "vue-router";
-
-const router = useRouter();
+import { settingsOutline, personOutline } from "ionicons/icons";
```
Comment (amansinghbais) [inline src/components/Tabs.vue]: Arrange alphabetically.

File: src/views/groups.vue [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script setup lang="ts">
+import { IonContent, IonHeader, IonMenuButton, IonPage, IonTitle, IonToolbar, onIonViewWillEnter } from "@ionic/vue";
+import { translate } from "../i18n"
+import { useGroupStore } from '../store/groups';
+import { computed, ref } from "vue";
```
Comment (amansinghbais) [inline src/views/groups.vue]: Remove unused ref.

File: src/components/Tabs.vue [hunk 11]
```diff
@@ -2,8 +2,12 @@
   <ion-page>
     <ion-tabs>
       <ion-router-outlet></ion-router-outlet>
-      <ion-tab-bar slot="bottom" v-if="showFooter()">
-        <ion-tab-button tab="more" href="/tabs/settings">
+      <ion-tab-bar slot="bottom">
+        <ion-tab-button tab="group" href="/tabs/group">
+          <ion-icon :icon="albumsOutline" />
+          <ion-label>{{ translate("Group") }}</ion-label>
```
Comment (amansinghbais) [inline src/components/Tabs.vue]: Make the tab as Groups

File: src/router/index.ts [hunk 12]
```diff
@@ -32,19 +32,28 @@ const authGuard = (to: any, from: any, next: any) => {
...
   {
-    path: "/settings",
-    name: "Settings",
-    component: Settings,
-    beforeEnter: authGuard
+    path: '/tabs',
+    component: Tabs,
+    children: [
+      {
+        path: 'group',
```
Comment (amansinghbais) [inline src/router/index.ts]:
```suggestion
        path: 'groups',
```

File: src/services/GroupService.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
+import { useUserStore } from '@/store/user'
+import api from "@/api"
+import logger from '../logger';
```
Comment (amansinghbais) [inline src/services/GroupService.ts]: Why are we importing logger using ../ path notation instead of using @/

File: src/views/groups.vue [hunk 14]
```diff
@@ -0,0 +1,81 @@
+<template>
+  <ion-page>
+    <ion-header :translucent="true">
+      <ion-toolbar>
+        <ion-menu-button slot="start" />
+        <ion-title>Napita Flow Manager</ion-title>
```
Comment (amansinghbais) [inline src/views/groups.vue]: Translate static texts

File: src/views/groups.vue [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+          <div>
+            <ion-item lines="none">
+              <ion-label>
+                <ion-note class="overline">{{ CurrentProcessGroup.id }}</ion-note>
+                <h1>{{ CurrentProcessGroup.name }}</h1>
+              </ion-label>
+            </ion-item>
+          </div>
+          <hr />
+          <div v-if="currentProcessBygroupDetail.length > 0">
```
Comment (amansinghbais) [inline src/views/groups.vue]:
```suggestion
          <div v-if="currentProcessBygroupDetail.length">
```

File: src/views/groups.vue [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+            <ion-accordion-group>
+              <ion-radio-group >
```
Comment (amansinghbais) [inline src/views/groups.vue]: We can remove ion-radio-group since we are not using ion-radio here

File: src/views/groups.vue [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+</template>
+
+<script setup lang="ts">
+import { IonContent, IonHeader, IonMenuButton, IonPage, IonTitle, IonToolbar, onIonViewWillEnter } from "@ionic/vue";
+import { translate } from "../i18n"
+import { useGroupStore } from '../store/groups';
+import { computed, ref } from "vue";
+
+const groupStore = useGroupStore();
+const currentProcessBygroupDetail = computed(() => groupStore.getCurrentGroupProcesses)
```
Comment (amansinghbais) [inline src/views/groups.vue]:
```suggestion
const currentGroupProcesses = computed(() => groupStore.getCurrentGroupProcesses)
```

File: src/views/groups.vue [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+const CurrentProcessGroup = computed(() => groupStore.getCurrentGroup);
```
Comment (amansinghbais) [inline src/views/groups.vue]:
```suggestion
const currentProcessGroup = computed(() => groupStore.getCurrentGroup);
```

File: src/views/groups.vue [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+async function fetchProcessByGroups(id: string) {
+  await groupStore.fetchProcessByGroups(id);
+}
+
+async function onProcessGroupChange(group: any) {
+  const selectedProcessGroupId = group.id
+  const selectedProcessGroup = processGroups.value.find((group: any) => group.id === selectedProcessGroupId);
+  groupStore.setCurrentProcessGroup(selectedProcessGroup);
+  // Fetch and display process details for the selected group
+  await fetchProcessByGroups(selectedProcessGroupId);
```
Comment (amansinghbais) [inline src/views/groups.vue]: Directly call action here instead of calling function.

File: src/services/GroupService.ts [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+  const userStore = useUserStore();
+  const baseURL = userStore.getBaseUrl;
+  return api({
+    url: "flow/process-groups/root",
+    method: "GET",
+    baseURL,
+  })
+}
+
+const fetchProcessByGroups = async (root: any): Promise<any> => {
```
Comment (amansinghbais) [inline src/services/GroupService.ts]:
```suggestion
const fetchProcessByGroups = async (groupId: any): Promise<any> => {
```

File: src/store/groups.ts [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+    getCurrentGroupProcesses(state) {
+      return state.currentGroupProcesses.processes
+    },
+    getCurrentGroupconnections(state) {
+      return state.currentGroupProcesses.connections
+    }
+  },
+  actions: {
+    async fetchProcessGroups() {
+      try{
```
Comment (amansinghbais) [inline src/store/groups.ts]:
```suggestion
      try {
```

File: src/views/groups.vue [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+    </ion-header>
+
+    <ion-content>
+      <div class="group">
+        <aside>
+          <h1>{{ translate("Process Groups") }}</h1>
+          <ion-list>
+            <ion-item v-for="group in processGroups" :key="group?.id" button detail @click="onProcessGroupChange(group)">
+              <ion-label :color="group.id === currentProcessGroup?.id ? 'primary' : ''">
+                <!-- <p class="overline">{{ group.id }}</p> -->
```
Comment (amansinghbais) [inline src/views/groups.vue]: Why is this line commented?

File: src/router/index.ts [hunk 23]
```diff
@@ -32,19 +32,28 @@ const authGuard = (to: any, from: any, next: any) => {
...
     path: '/',
-    redirect: '/settings'
+    redirect: '/tabs/settings'
   },
   {
     path: '/login',
     name: 'Login',
     component: Login,
     beforeEnter: loginGuard
+
```
Comment (amansinghbais) [inline src/router/index.ts]: Remove unnecessary space.

File: src/views/groups.vue [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+              <ion-label>
+                <ion-note class="overline">{{ currentProcessGroup.id }}</ion-note>
+                <h1>{{ currentProcessGroup.name }}</h1>
+              </ion-label>
+            </ion-item>
+          </div>
+          <hr />
+          <div v-if="currentGroupProcesses.length">
+            <ion-accordion-group>
+                <ion-accordion v-for="(process, index) in currentGroupProcesses" :key="index" >
```
Comment (amansinghbais) [inline src/views/groups.vue]: improve intendation.

File: src/views/groups.vue [hunk 25]
```diff
@@ -0,0 +1,75 @@
+<template>
+  <ion-page>
+    <ion-header :translucent="true">
+      <ion-toolbar>
+        <ion-menu-button slot="start" />
```
Comment (amansinghbais) [inline src/views/groups.vue]: Remove this ion-menu-button if not used.

File: src/views/groups.vue [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+                <ion-item slot="header">
+                  <ion-label class="ion-text-wrap">
+                    {{ process.name }}
+                  </ion-label>
+                </ion-item>
+              </ion-accordion>
+            </ion-accordion-group>
+          </div>
+          <div v-else class="empty-state" >
+            <p>No process found for {{ currentProcessGroup.name }} .</p>
```
Comment (ymaheshwari1) [inline src/views/groups.vue:43]: Add support for internationalization.

File: src/views/groups.vue [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+import { IonAccordion, IonAccordionGroup, IonContent, IonHeader, IonItem, IonLabel, IonList, IonMenuButton, IonNote, IonPage, IonTitle, IonToolbar, onIonViewWillEnter } from "@ionic/vue";
+import { translate } from "@/i18n"
+import { useGroupStore } from '@/store/groups';
+import { computed } from "vue";
+
+const groupStore = useGroupStore();
+
+const currentGroupProcesses = computed(() => groupStore.getCurrentGroupProcesses)
+const currentProcessGroup = computed(() => groupStore.getCurrentGroup);
+const processGroups = computed(() =>groupStore.getProcessGroups);
```
Comment (ymaheshwari1) [inline src/views/groups.vue:61]:
```suggestion
const processGroups = computed(() => groupStore.getProcessGroups);
```

File: src/store/groups.ts [hunk 28]
```diff
@@ -0,0 +1,90 @@
...
+          processGroups.sort((a:any, b: any) => a.name.localeCompare(b.name));
+          this.processGroups = processGroups;
+        } else {
+          throw resp.data
+        }
+      } catch (error) {
+        logger.error(error)
+      }
+    },
+    async fetchProcessByGroups(id: any) {
```
Comment (ymaheshwari1) [inline src/store/groups.ts:55]:
```suggestion
    async fetchProcessByGroup(id: string) {
```

File: src/store/groups.ts [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+          this.currentGroupProcesses.processes = processes;
+          this.currentGroupProcesses.connections = connections;
+        } else {
+          throw resp.data
+        }
+      } catch (error) {
+        logger.error(error)
+      }
+    },
+    setcurrentProcessGroup(processGroup: any) {
```
Comment (ymaheshwari1) [inline src/store/groups.ts:85]:
```suggestion
    setCurrentProcessGroup(processGroup: any) {
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Pull request merged**